clash-lib/src/Clash/Netlist/Util.hs

```diff
@@ -1000,6 +1004,41 @@ prefixParent parent (PortName p)        = PortName (parent <> "_" <> p)
 prefixParent parent (PortProduct "" ps) = PortProduct parent ps
 prefixParent parent (PortProduct p ps)  = PortProduct (parent <> "_" <> p) ps
 
+mkAssign :: Identifier -> HWType -> Expr -> [Declaration]
+mkAssign id_ hwty expr = [NetDecl Nothing id_ hwty, Assignment id_ expr]
+
+convPrimitiveType :: HWType -> a -> NetlistMonad a -> NetlistMonad a
+convPrimitiveType hwty a action = do
+  b <- Lens.use backend
+  let kind = case b of {SomeBackend s -> State.evalState (hdlHWTypeKind hwty) s}
+  case kind of
+    UserType -> action
+    SynonymType -> pure a
+    PrimitiveType -> pure a
+
+toPrimitiveType
+  :: Identifier
+  -> HWType
+  -> NetlistMonad ([Declaration], Identifier, Expr, HWType)
+toPrimitiveType id0 hwty0 = convPrimitiveType hwty0 dflt $ do
+  id1 <- mkUniqueIdentifier Basic id0
```

Comment:
Any reason to use `Basic` instead of `Extended` identifiers here? Extended identifiers are better at preserving original Haskell names, and should be preferred. We only use `Basic` identifiers in specific cases, such as: label names, and GenSym template names.

Reply:
No! I'd been meaning to ask what to do, but I forgot.